src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -473,7 +473,11 @@ private ValueTask StreamDisposeAsync(Stream stream)
                     this.logger.LogImageServed(imageContext.GetDisplayUrl(), key);
 
                     // When stream is null we're sending from the cache.
-                    await imageContext.SendAsync(stream ?? await cacheResolver.OpenReadAsync(), metadata);
+                    using (Stream output = stream ?? await cacheResolver.OpenReadAsync())
```

Comment:
This will also mean that you will dispose the input stream. Don't think we should?

Reply:
Ah no, we always copy the input stream so we have control.

https://github.com/SixLabors/ImageSharp.Web/blob/8f67aae9d314a6fde96aae856027a655d6cbd2fa/src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs#L296-L302

Reply:
But later we will call `StreamDisposeAsync`?

Reply:
Yeah but I'd consider that safe. `Dispose`, by design should allow multiple calls and `RecyclableMemoryStream` certainly allows it. The pipeline is going through a bit of a refactor just now via #160 so it's gonna change anyway.

Reply:
If the pipeline was working how it should this would be an issue, but it’s not, so that outsream is single use, so as long as it handles dispose being called twice ok it should be ok

Reply:
Lot of shoulds in that statement :)

 makes me think it might be better to handle the two types of streams separately, at least until the refactor removes it